ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -852,8 +858,12 @@ protected void maybeProcessCallbacks() {
               routerMetrics.getFirstDataChunkLatencyMs.update(System.currentTimeMillis() - submissionTimeMs);
             }
             routerMetrics.getDataChunkLatencyMs.update(System.currentTimeMillis() - initializedTimeMs);
-            chunkIndexToBuf.put(chunkIndex, filterChunkToRange(result.getDecryptedBlobContent()));
-            numChunksRetrieved.incrementAndGet();
+            ByteBuf decryptedContent = result.getDecryptedBlobContent();
+            ByteBuf decompressedContent = decompressContent(decryptedContent);
+            if (decompressedContent != null) {
+              chunkIndexToBuf.put(chunkIndex, filterChunkToRange(decompressedContent));
+              numChunksRetrieved.incrementAndGet();
```

Comment:
Curious, in case of exception in decompress (therefore decompressedContent=null), do we still need to run `numChunksRetrieved.incrementAndGet()` or not?

Reply:
numChunksRetrieved is used to send back to client, not for statistic purpose.  So unless the content is available, we should not increment numChunksRetrieved.  Decrypt and decompress is part of the retrieving steps, so unless both steps succeeded, any failure should be considered as retrieval failure.